### PR TITLE
[SYCL] Fix Kernel Compiler OpenCLC_Supports_Extension

### PR DESCRIPTION
**Pull request by bso-intel**
The optional parameter `VersionPtr` is `nullptr` by default. But, the code assumes it is a valid pointer.
So, the fix is to guard with an `if` conditional phrase.

### CODE REVIEW

File: sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp [hunk 1]
```diff
@@ -340,27 +340,31 @@ bool OpenCLC_Supports_Extension(
         "trouble parsing query returned from CL_DEVICE_EXTENSIONS_WITH_VERSION "
         "- extension not followed by colon (:)");
   }
-  colon++; // move it forward
 
-  size_t space = ExtensionByVersionLog.find(' ', colon); // could be npos
+  // Note that VersionPtr is an optional parameter in
+  // ext_oneapi_supports_cl_extension().
+  if (VersionPtr) {
```
Comment (aelovikov-intel) [inline sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp:346]: This should be an early return idiom for readability.

**Review by aelovikov-intel — CHANGES_REQUESTED** [review]
Tests are missing for a bugfix PR.

File: sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp (hunk 1 above)
Comment (bso-intel) [reply sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp:346]: If VersionPtr is not nullptr, setting the version information is needed before we return.
I think the current code is more readable than redundantly adding "return true".
```
if (!versionPtr)
  return true;
// all the content inside the if-condition
return true;
```

Comment (aelovikov-intel) [reply sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp:346]: That is *NOT* up to the debate, our coding style requires it be an early return, see https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code.

**Comment by bso-intel** [comment]
> Tests are missing for a bugfix PR.

Added tests ext_oneapi_support_cl_extension() without versionPtr parameter set.

File: sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp (hunk 1 above)
Comment (bso-intel) [reply sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp]: Ok. changed.

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**